### ISSUE TITLE
How make docker host respond on port 2375 on Centos 7

### ISSUE THREAD

**Issue by eliassal** [docker/for-win#318]
I have docker on Centos 7, just upgraed it to 1.10.3 using yum. I have a Docker file in etc/sysconfig.
I am trying to make docker listen on 0.0.0.0:2375 or machineIP:2375 and not 127.0.0.1
I added the following line 
 DOCKER_OPTS="-H tcp://127.0.0.1:2375 -H unix:///var/run/docker.sock" 
restarted docker, nothing happens, I don't see anything listening on port 2375. I stopped docker and rerun it with the command 
 sudo docker "-H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock
getting ssh error : could not resolve host name unix:///var/run/docker.sock, service unkown.

I searched  and found som references to a file docker.service which is in my usr/lib/systemd/system with very basic information
EnvironmentFile=.....
Env....

So can you please let me know in very simple straight forward how to acheive this on Centos. I went through  "Control and configure Docker with systemd" at
https://docs.docker.com/engine/admin/systemd/
but it is a little bit confusing for a person who is not expert in linux
Thanks in advance. Note : I am trying this because I was not able to aceive this on Docker for windows, thread I opened yesterday #314

**Comment by rneugeba** [comment]
I'm closing this issue as it is not related to Docker for Windows.

You need to start the docker *daemon* to listen on port 2375, so something like:
```
dockerd -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock
```
(note the `d` and no quotes compared to your command line)

then you need to tell the docker client how to connect to the server. A common way to do so is using the `DOCKER_HOST` environment variable. Like
```shell
export DOCKER_HOST=tcp://192.168.99.101:2375
docker info
```
Note, you need to change `192.168.99.101` to the IP address of your server.

You also need to make sure that the server does not block port 2375 with its firewall.

In general it's a bad idea to open up un-encrypted ports. See this on how to use TLS: https://docs.docker.com/engine/security/https/

**Comment by eliassal** [comment]
Ok thanks, in the future I will post to correct forum. However, I stopped the docker server with 
systemctl docker stop
which went fine but when I try 
sudo dockerd -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock
I get back 
sudo: dockerd: command not found

Thanks again for your help

**Comment by eliassal** [comment]
Also, can you please share the url for the forum for docker for centos
Thanks

**Comment by rneugeba** [comment]
I don't know about centos and where dockerd is installed and how it is started. You could try the general discussion or the DockerEngine forums on https://forums.docker.com/ or stackoverflow. there may also be centos specific fora

**Comment by eliassal** [comment]
meugeba, I have successfully started docker on a win 2k16 using a daemon.json file with only
{
    "graph": "E:\\DockerMachines",
	"hosts": ["tcp://192.168.1.34:2375", "npipe://"]

}
I checked it is listening on port 2375. However, whatever run or pull command I run, I get

docker pull nanori/jenkins-windows-slave
Using default tag: latest
Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

Any idea? Thanks for your help.
